src/pf-tooltip/pf-tooltip.component.js

```diff
@@ -0,0 +1,304 @@
+      this.setAttribute('container', value);
+    }
+  }
+
+  /**
+   * Get the animation duration
+   *
+   * @returns {string} The animation duration
+   */
+  get duration () {
+    return this._duration;
+  }
+
+  /**
+   * Set the animation duration
+   *
+   * @param {string} value The animation duration
+   */
+  set duration (value) {
+    if (this._duration !== value) {
+      this._duration = value;
+      this.setAttribute('duration', value);
+    }
+  }
+
+  /**
+   * Get the animation delay
+   *
+   * @returns {string} The animation delay
+   */
+  get delay () {
+    return this._duration;
+  }
+
+  /**
+   * Set the animation delay
+   *
+   * @param {string} value The animation delay
+   */
+  set delay (value) {
+    if (this._delay !== value) {
+      this._delay = value;
+      this.setAttribute('delay', value);
+    }
+  }
+
+  /**
+   * Get the placement position
+   *
+   * @returns {string} The placement position left, top, bottom, right
+   */
+  get placement () {
+    return this._placement;
+  }
+
+  /**
+   * Set placement position
+   *
+   * @param {string} value The placement position left, top, bottom, right
+   */
+  set placement (value) {
+    if (this._placement !== value) {
+      this._placement = value;
+      this.setAttribute('placement', value);
+    }
+  }
+
+  /**
+   * Get the target element
+   *
+   * @returns {string} The target element
+   */
+  get target () {
+    return this._target;
+  }
+
+  /**
+   * Set target element
+   *
+   * @param {string} value The target element
+   */
+  set target (value) {
+    if (this._target !== value) {
+      this._target = value;
+      this.setAttribute('target', value);
+    }
+  }
+
+  /**
+   * The open event handler
+   * @param e
+   */
+  open (e) {
+    clearTimeout(this._timer);
+    this._timer = setTimeout( () => {
+      if (this.tooltip === null) {
+        this.createTooltip();
+        this.styleTooltip();
+        this.showTooltip();
+      }
+    }, 20 );
+  }
+
+  /**
+   * The close event handler
+   * @param e
+   */
+  close (e) {
+    clearTimeout(this._timer);
+    this._timer = setTimeout( () => {
+      if (this.tooltip && this.tooltip !== null) {
+        pfUtil.removeClass(this.tooltip, 'in');
+        setTimeout(() => {
+          this.removeTooltip();
+        }, this._duration);
+      }
+    }, this._delay + this._duration);
+  }
+
+  /**
+   * Removes the tooltip
+   */
+  removeTooltip () {
+    this.tooltip && this.container.removeChild(this.tooltip);
+    this.tooltip = null;
+  }
+
+  /**
+   * Creates the tooltip
+   */
+  createTooltip () {
+    let clone = document.importNode(this._template.content, true);
+    let tooltipInner = clone.querySelector('.tooltip-inner');
+
+    //set tooltip content
+    tooltipInner.innerHTML = this.content;
+
+    //append to the container
+    this.container.appendChild(clone);
+
+    //set reference to appended node
+    let tooltips = this.container.querySelectorAll('.tooltip');
```

Comment:
This selector name (.tooltip) may be too generic? It's possible that this could match something else in the page. That is, considering container could be the body tag.

The '.tooltip-inner' selector looks ok because it's found under this._template.content.